targets/simple_switch/simple_switch.cpp

```diff
@@ -520,18 +517,20 @@ SimpleSwitch::ingress_thread() {
     }
 
     // INGRESS CLONING
-    if (clone_spec) {
+    if (clone_mirror_session_id) {
```

Comment:
Antonin, one nitpick question for you I wasn't sure about.  The `if` condition here is actually slightly different than the original.  To make it completely equivalent to the original condition, I could replace it with: `if (clone_mirror_session_id || clone_field_list)`, but I was guessing that perhaps clone mirror session ids are restricted to be non-0.  Is that true?

Reply:
And if they can be 0, then it was probably slightly buggy to begin with, if P4_16 clone() was implemented by using a field_list_id of 0.

Reply:
Actually that's an important question :)
I believe the assumption is that the mirror session id is non-0. Even though the P4Runtime spec states that 0 is a valid value, the reference P4Runtime implementation will currently not let you use 0 (https://github.com/p4lang/PI/blob/master/proto/frontend/src/pre_clone_mgr.h#L69). So maybe documenting this (in the code / the simple_switch documentation) is important.

Reply:
If you want, I can with these changes add another "architecture private" field that is a 1-bit boolean flag remembering whether a clone operation was ever performed, and use that in the `if` conditions.  Then it will cleanly support a clone/mirror session id of 0.

Reply:
Up to you, you can also use the msb of the 16-bit register
personally I don't think it's a big deal since there is no regression

Reply:
I will go for using msb of 16-bit value in next commit.  Seems worth cleanly supporting clone/mirror session id of 0 as a small enhancement.  Then no need to document that 0 is not supported :)